lms/views/api/canvas/authorize.py

```diff
@@ -20,25 +20,33 @@
 
 @view_defaults(request_method="GET", route_name="canvas_oauth_callback")
 class CanvasAPIAuthorizeViews:
+
+    #: The Canvas API scopes that we request.
+    scopes = (
+        "url:GET|/api/v1/courses/:course_id/files",
+        "url:GET|/api/v1/files/:id/public_url",
+    )
+
     def __init__(self, request):
         self.request = request
-        self.ai_getter = request.find_service(name="ai_getter")
```

Comment:
`request.find_service(name="ai_getter")` here in the `__init__()` was crashing because, when the view is `CanvasAPIAuthorizeViews.oauth2_redirect_error()`, `request.lti_user` is `None` and this causes `ai_getter`'s service factory to crash. Moved this line into the one specific view method that actually uses `ai_getter` instead. That fixes the crash